internal/network/utils.go

```diff
@@ -161,25 +162,41 @@ func GetClusterNetworkCidrs(cluster *common.Cluster) (ret []string) {
 	return
 }
 
-func CidrToAddressFamily(cidr string) (string, error) {
+func CidrToAddressFamily(cidr string) (AddressFamily, error) {
 	_, _, err := net.ParseCIDR(cidr)
 	if err != nil {
-		return "", err
+		return 0, err
 	}
 	if strings.Contains(cidr, ":") {
-		return "IPv6", nil
+		return IPv6, nil
 	}
-	return "IPv4", nil
+	return IPv4, nil
 }
 
-func CidrsToAddressFamilies(cidrs []string) (ret []string, err error) {
+func CidrsToAddressFamilies(cidrs []string) (ret []AddressFamily, err error) {
 	for _, cidr := range cidrs {
-		var af string
+		var af AddressFamily
 		af, err = CidrToAddressFamily(cidr)
 		if err != nil {
 			return nil, err
 		}
 		ret = append(ret, af)
 	}
+	sort.Slice(ret, func(i, j int) bool {
+		return ret[i] < ret[j]
+	})
+	return
+}
+
+func CanonizeAddressFamilies(slice []AddressFamily) (ret []AddressFamily) {
```

Comment:
Maybe overengineering, but something like below?

```
set := make(map[AddressFamily]bool)
for _, val := range slice {
  set[val] = true
}
for key, val := range set {
  if val {
    ret = append(ret, key)  
  }
}
```

this way you make sure that the list you return will never have duplicate values (and from the context I get that this is what you want here)

Reply:
Why is it better ?

Reply:
>Also, it is not the same

Can we then get a comment for this function what it's supposed to do? As I see the current implementation, it goes by sliding window over every neighbouring field of `slice` (so always compares N-1 and N) and keeps one of them if both are equal. So for the following slice `v4, v4, v4, v4, v6` it will return `v4, v6`. But if your initial slice is `v4, v6, v4, v6` then value returned will be `v4, v6, v4, v6`. So either we need to handle that or make sure that what is passed to this function is always sorted

Reply:
Will add comment

Reply:
The intention for this function is to keep the existing order